### PR TITLE
[ISSUE-633][FEATURE] Support provider user identity by customized class and keep LifecycleManager and ShuffleClient user identity consistence

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @FMX @RexXiong

File: client-spark/shuffle-manager-2/src/main/scala/org/apache/spark/shuffle/rss/RssShuffleManager.scala [hunk 1]
```diff
@@ -61,7 +62,12 @@ class RssShuffleManager(conf: SparkConf) extends ShuffleManager with Logging {
         if (lifecycleManager.isEmpty) {
           val metaSystem = new LifecycleManager(appId, rssConf)
           lifecycleManager = Some(metaSystem)
-          rssShuffleClient = Some(ShuffleClient.get(metaSystem.self, rssConf))
+          rssShuffleClient = Some(ShuffleClient.get(
+            metaSystem.self,
+            rssConf,
+            lifecycleManager.map(_.getUserIdentifier()).getOrElse(UserIdentifier(
```
Comment (waitinfuture) [inline client-spark/shuffle-manager-2/src/main/scala/org/apache/spark/shuffle/rss/RssShuffleManager.scala:68]: lifecycleManager is guaranteed to have value here

File: common/src/main/scala/com/aliyun/emr/rss/common/identity/IdentityProvider.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  val DEFAULT_TENANT_ID = "default"
+
+  def instantiate(conf: RssConf): IdentityProvider = {
+    val className = RssConf.identityProviderClass(conf)
+    logDebug(s"Creating identity provider $className")
+    val clazz = Class.forName(
+      className,
+      true,
+      Thread.currentThread().getContextClassLoader).asInstanceOf[Class[IdentityProvider]]
+    // First try the constructor with arguments (jobId: String, outputPath: String,
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/com/aliyun/emr/rss/common/identity/IdentityProvider.scala:38]: Yea

File: client-spark/shuffle-manager-2/src/main/scala/org/apache/spark/shuffle/rss/RssShuffleManager.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply client-spark/shuffle-manager-2/src/main/scala/org/apache/spark/shuffle/rss/RssShuffleManager.scala]: Yea

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**